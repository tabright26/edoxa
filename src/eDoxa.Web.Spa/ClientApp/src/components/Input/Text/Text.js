import React from "react";
import { Input, FormFeedback } from "reactstrap";

const TextInput = ({ input, label, type, meta: { touched, error } }) => (
  <>
    <Input {...input} placeholder={label} type={type} bsSize="sm" invalid={touched && error} />
    <FormFeedback>{error}</FormFeedback>
  </>
);

export default TextInput;
