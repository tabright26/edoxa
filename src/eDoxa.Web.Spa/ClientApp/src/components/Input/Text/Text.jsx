import React from "react";
import { Input, FormFeedback } from "reactstrap";

const TextInput = ({ input, label, type = "text", meta: { touched, error }, ...attributes }) => (
  <>
    <Input {...input} {...attributes} placeholder={label} type={type} invalid={touched && error} />
    <FormFeedback>{error}</FormFeedback>
  </>
);

export default TextInput;
