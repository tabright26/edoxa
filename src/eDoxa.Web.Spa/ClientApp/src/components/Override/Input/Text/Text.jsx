import React from "react";
import { Input, FormFeedback } from "reactstrap";

const TextInput = ({ input, label, meta: { touched, error }, ...attributes }) => (
  <>
    <Input bsSize="sm" {...input} {...attributes} placeholder={label} invalid={touched && error} />
    <FormFeedback>{error}</FormFeedback>
  </>
);

export default TextInput;
