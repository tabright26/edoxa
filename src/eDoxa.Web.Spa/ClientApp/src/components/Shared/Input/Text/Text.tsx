import React, { Fragment } from "react";
import { Input, FormFeedback } from "reactstrap";

const TextInput = ({
  formGroup: FormGroup = Fragment,
  input = null,
  label = null,
  disabled,
  meta = null,
  ...attributes
}) => (
  <FormGroup>
    <Input
      {...input}
      {...attributes}
      bsSize="sm"
      disabled={disabled}
      placeholder={label}
      invalid={!disabled && meta.touched && !!meta.error}
    />
    {meta && <FormFeedback>{meta.error}</FormFeedback>}
  </FormGroup>
);

export default TextInput;
