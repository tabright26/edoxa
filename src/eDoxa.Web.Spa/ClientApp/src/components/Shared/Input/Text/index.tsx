import React, { FunctionComponent, Fragment } from "react";
import { Input, FormFeedback, Label } from "reactstrap";

export const Text: FunctionComponent<any> = ({
  input = null,
  label = null,
  placeholder = null,
  disabled,
  size = "sm",
  meta = null,
  formGroup: FormGroup = Fragment,
  ...props
}) => (
  <FormGroup>
    {label && <Label>{label}</Label>}
    <Input
      {...input}
      {...props}
      placeholder={placeholder}
      bsSize={size}
      disabled={disabled}
      invalid={!disabled && meta.touched && !!meta.error}
    />
    {meta && <FormFeedback>{meta.error}</FormFeedback>}
  </FormGroup>
);
