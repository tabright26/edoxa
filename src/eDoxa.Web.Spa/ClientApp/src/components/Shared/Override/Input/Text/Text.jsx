import React, { Fragment } from "react";
import { Input, FormFeedback } from "reactstrap";

const TextInput = ({ formGroup: FormGroup = Fragment, input, label, disabled, meta, ...attributes }) => (
  <FormGroup>
    <Input bsSize="sm" {...input} {...attributes} disabled={disabled} placeholder={label} invalid={!disabled && meta.touched && meta.error} />
    {meta ? <FormFeedback>{meta.error}</FormFeedback> : null}
  </FormGroup>
);

export default TextInput;
