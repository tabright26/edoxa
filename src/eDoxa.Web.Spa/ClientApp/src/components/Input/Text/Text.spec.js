import React from "react";
import Text from ".";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Text meta={{ touched: true }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
