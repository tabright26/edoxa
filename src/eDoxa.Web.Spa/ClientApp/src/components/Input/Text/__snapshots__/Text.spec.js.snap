// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <input
    className="form-control-sm form-control"
    type="text"
  />,
  <div
    className="invalid-feedback"
  />,
]
`;
